python/lsst/resources/s3utils.py

```diff
@@ -202,26 +220,74 @@ def getS3Client() -> boto3.client:
     if botocore is None:
         raise ModuleNotFoundError("Could not find botocore. Are you sure it is installed?")
 
-    endpoint = os.environ.get("S3_ENDPOINT_URL", None)
-    if not endpoint:
-        endpoint = None  # Handle ""
+    if profile is None:
+        endpoint = os.environ.get("S3_ENDPOINT_URL", None)
+        if not endpoint:
+            endpoint = None  # Handle ""
+    else:
+        var_name = f"LSST_RESOURCES_S3_PROFILE_{profile}"
+        endpoint = os.environ.get(var_name, None)
+        if not endpoint:
+            raise RuntimeError(
+                f"No configuration found for requested S3 profile '{profile}'."
```

Comment:
boto3 can retrieve profiles from the `AWS_SHARED_CREDENTIALS_FILE` if all of them have the same `S3_ENDPOINT_URL`.  I think we need to allow this capability.

Reply:
So you're saying you want support for passing the profile name to boto for looking up the credentials?  As-is it will already retrieve the default credentials from there if you specify only the endpoint URL as either `LSST_RESOURCES_S3_PROFILE_profile=http://onlytheendpoint-with-no-credentials.com` or `S3_ENDPOINT_URL=http://onlytheendpoint.com`.

Reply:
I was thinking that since we already can have multiple credentials in the `AWS_SHARED_CREDENTIALS_FILE` that we should continue to use them, especially if they all share a single `S3_ENDPOINT_URL` but even if they do not.  So if the `LSST_RESOURCES_S3_PROFILE_{profile}` env var doesn't exist, or if it lacks `{access}:{secret}`, pass the profile name to boto.